bin/jobemissions

```diff
@@ -59,32 +82,65 @@ food_d = {
+        epilog="The carbon density for Scope 2 emissions is taken as the value for the location of CREATE HPC at the start of the job from carbonintensity.org.uk. Scope 3 emissions are computed based on the amount of resource used by the job (e.g. kgCO2e/nodeh) and are calculated as the total Scope 3 emissions from the service divided byt the total number of nodeh available over the projected lifetime of the service.\n",
         )
 parser.add_argument("jobid", type=str, nargs=1, help="JobID of the Slurm job to compute emissions for.")
 parser.add_argument("--json", dest="jsonOutput", action="store_true", default=False, help="Output in JSON format rather than human-readable text.")
 parser.add_argument("--comparison", type=str, dest="compare", action="store", default="none", help='Specify emissions comparisons to print as a comma-separated list. Valid vales "food", "other", "none". "none" suppresses reporting of emissions comparisons. Default is "none".')
-parser.add_argument("--cisource", type=str, dest="cisource", action="store", default="cache", help='Specify how to source the carbon intensity value for indicative Scope 2 emissions. "cache" gets values from cache on ARCHER2 to avoid flooding carbonintensity.org.uk with requests and allow use on compute nodes. "ciwebsite" fetches values direct from carbonintensity.org.uk website. Default is "cache".')
+parser.add_argument("--cisource", type=str, dest="cisource", action="store", default="cache", help='Specify how to source the carbon intensity value for indicative Scope 2 emissions. "cache" gets values from cache on CREATE HPC to avoid flooding carbonintensity.org.uk with requests and allow use on compute nodes. "ciwebsite" fetches values direct from carbonintensity.org.uk website. Default is "cache".')
 args = parser.parse_args()
 
 jobid = args.jobid[0]
 
 # Get job details
 job_energy = 0
-sacct_cmd = subprocess.run(["sacct", "-n", "-X", "--format=Start,NNode,ElapsedRaw,Account%50,ConsumedEnergyRaw", f"--job={jobid}"], stdout=subprocess.PIPE, check=False)
-job_details = sacct_cmd.stdout.decode("utf-8").strip()
+sacct_cmd = subprocess.run(["sacct", "-n", "-X", "--format=Start,NNode,ElapsedRaw,NodeList,AllocCPUS,ConsumedEnergyRaw", f"--job={jobid}"], stdout=subprocess.PIPE, check=False)
+job_details = sacct_cmd.stdout.decode("utf-8").strip()  #makes slurm output readable
+
+if sacct_cmd.returncode != 0: 
+    print(f"Error: 'sacct' command failed for job ID {jobid}")
+    sys.exit(1)
+if not job_details:
+    print(f"Error: No data returned for job ID {jobid}")
+    sys.exit(1)
+tokens = job_details.split()
+if len(tokens) < 6:
+    print(f"Error: Incomplete data returned for job ID {jobid}")
+    sys.exit(1)
```

Comment:
Could this be turned into a function that takes a job id as input and returns the tokens?

Then you could update the tests in `test_tokens.py` to use that function.

You could add an extra test in that file to assert that you get an error if you provide an invalid job id, as well as checking that the tokens are returned correctly with a real job id.